django_ibmi/base.py

```diff
@@ -170,139 +171,53 @@ def __is_connection(self):
+            conn_params['user'] = settings_dict['USER']
+        if 'PASSWORD' in settings_dict:
+            conn_params['password'] = settings_dict['PASSWORD']
+
+        if 'OPTIONS' in settings_dict:
+            conn_params.update(settings_dict['OPTIONS'])
+
+        allowed_opts = {'system', 'user', 'password', 'autocommit',
+                        'readonly',
+                        'timeout', 'database', 'use_system_naming',
+                        'library_list', 'current_schema'
+                        }
+
+        if not allowed_opts.issuperset(conn_params.keys()):
+            raise ValueError("Option entered not valid for "
+                             "IBM i Access ODBC Driver")
 
-        if 'CURRENTSCHEMA' in settings_dict:
-            database_schema = settings_dict['CURRENTSCHEMA']
-            if isinstance(database_schema, str):
-                kwargs['currentschema'] = database_schema
-
-        if 'SECURITY' in settings_dict:
-            database_security = settings_dict['SECURITY']
-            if isinstance(database_security, str):
-                kwargs['security'] = database_security
-
-        if 'SSLCLIENTKEYDB' in settings_dict:
-            database_sslclientkeydb = settings_dict['SSLCLIENTKEYDB']
-            if isinstance(database_sslclientkeydb, str):
-                kwargs['sslclientkeydb'] = database_sslclientkeydb
-
-        if 'SSLCLIENTKEYSTOREDBPASSWORD' in settings_dict:
-            database_sslclientkeystoredbpassword = settings_dict['SSLCLIENTKEYSTOREDBPASSWORD']
-            if isinstance(database_sslclientkeystoredbpassword, str):
-                kwargs['sslclientkeystoredbpassword'] = database_sslclientkeystoredbpassword
-
-        if 'SSLCLIENTKEYSTASH' in settings_dict:
-            database_sslclientkeystash = settings_dict['SSLCLIENTKEYSTASH']
-            if isinstance(database_sslclientkeystash, str):
-                kwargs['sslclientkeystash'] = database_sslclientkeystash
-
-        if 'SSLSERVERCERTIFICATE' in settings_dict:
-            database_sslservercertificate = settings_dict['SSLSERVERCERTIFICATE']
-            if isinstance(database_sslservercertificate, str):
-                kwargs['sslservercertificate'] = database_sslservercertificate
-
-        return kwargs
+        try:
+            conn_params['Naming'] = \
+                str(util.strtobool(conn_params['use_system_naming']))
+        except (ValueError, KeyError):
+            conn_params['Naming'] = '0'
+
+        if 'current_schema' in conn_params or 'library_list' in conn_params:
+            conn_params['DefaultLibraries'] = \
+                conn_params.pop('current_schema', '') + ','
+            if isinstance(conn_params["DefaultLibraries"], str):
+                conn_params['DefaultLibraries'] += \
+                    conn_params.pop('library_list', '')
+            else:
+                conn_params['DefaultLibraries'] += ','.join(
+                    conn_params.pop('library_list', ''))
```

Comment:
Shouldn't this be checking library_list? We already know that DefaultLibraries is a string, since we appended a comma to it in the last line.

Maybe something like this:
```suggestion
            library_list = conn_params.pop('library_list', '')
            if isinstance(library_list, str):
                conn_params['DefaultLibraries'] += library_list
            else:
                conn_params['DefaultLibraries'] += ','.join(library_list)
```